codice/ddf
# DDF-04346 - Preserving Original Query Structure For Advanced Queries

#### What does this PR do?
One common complaint we have received is the lack of preservation pertinent to queries being saved from Intrigue.

**Two Main Issues:**

    * Queries saved from a keyword re-render as a polygon or multipolygon (keyword metadata is lost and not preserved)

    * Queries saved from bounding box re-render as polygon (bbox corner points are not preserved and fail to reconstruct the correct bbox mode)


This user experience is strange and causes a lot of confusion due to lack of preserved information related to the original query.
#### Who is reviewing it? 
@Bdthomson 
@bdeining 
@middlets719 
@adimka 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@troymohl 
@millerw8 
@djblue 
@andrewkfiedler 

#### How should this be tested?
Create queries for all the location types using the advanced query builder, persist them, refresh, ensure that bbox and keywords persisted and reloaded correctly. 

#### Any background context you want to provide?
#### What are the relevant tickets?
Fixes: #4346 

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
